src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -413,7 +414,9 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
 
   if ( mPostgresqlVersion >= 90000 )
   {
-    LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET application_name='QGIS'" ) );
+    QgsSettings settings;
+    const QString applicationName = settings.value( QStringLiteral( "PostgreSQL/application_name" ), PG_APPLICATION_NAME, QgsSettings::Providers ).toString();
+    LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET application_name='%s'" ).arg( applicationName ) );
     LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET extra_float_digits=3" ) );
```

Comment:
`LoggedPQexecNR( "QgsPostgresConn", QStringLiteral( "SET application_name='%s';SET extra_float_digits=3" ).arg( applicationName ) );`

Combining several SQL queries together will slightly reduce the negative impact of network delays on the speed of QGIS with the database.
Downside: The code becomes a bit harder to understand.